scripts/HighNoon.js

```diff
@@ -0,0 +1,401 @@
+      sessionStore[channelId].gameState.elapsed = sessionStore[channelId].gameState.countdownValue;
+      if (sessionStore[channelId].gameState.timeToDraw()) {
+        msg.reply('Session ready state has been set to true. Players may now proceed to shoot.');
+      } else {
+        msg.reply('Uh oh! Session\'s ready state cannot be set to true. Please check the code.');
+      }
+    } else {
+      msg.reply('Sorry! A session has not been made yet, please type in &highnoon to start a session.');
+    }
+  });
+
+  robot.hear(/BANG!/i, function (msg) {
+    var person = messageHelper.getPerson(msg);
+    var channelId = messageHelper.getChannelId(msg);
+    // check if a session exists
+    if (sessionStore[channelId]) {
+      // check if the session already has all the players
+      if (sessionStore[channelId].gameState.sessionIsFull()) {
+        // check if the person is one of the players and if the duel is ongoing
+        if (sessionStore[channelId].gameState.players[person].getName() === person && !sessionStore[channelId].allShotsFired) {
+          // check if it is time to draw
+          if (sessionStore[channelId].gameState.timeToDraw()) {
+            msg.reply(person + ' shot first! Everyone else falls dead. ' + person + ' wins!');
+            for (var player in sessionStore[channelId].gameState.players) {
+              if (sessionStore[channelId].gameState.players[player].getName() !== person) {
+                sessionStore[channelId].gameState.players[player].gotShot();
+              }
+            }
+            sessionStore[channelId].gameState.duelIsDone();
+            removeSession(msg, channelId);
+          } else {
+            msg.reply(person + ' misfired! ' + person + ' is DEAD and out of the duel!');
+            sessionStore[channelId].gameState.players[person].gotShot();
+          } 
+        } else {
+          msg.reply('HEY! You\'re not part of this duel! Take a seat son and wait for your turn!');
+        }
+      } else {
+        msg.reply('Please wait for all players to enter the game first before shooting! (Players needed: ' + sessionStore[channelId].gameState.playersNeeded() + ')');
+      }
+    } else {
+      msg.reply('Sorry! A session has not started yet.  Please type in &highnoon to start a session.');
+    }
+  });
+
+  robot.hear(trigger.getTrigger('cancel'), function (msg) {
+    var channelId = messageHelper.getChannelId(msg);
+    removeSession(msg, channelId);
+  });
+
+  robot.hear(trigger.getTrigger('time'), function (msg) {
+    var channelId = messageHelper.getChannelId(msg);
+    if (sessionStore[channelId]) {
+      msg.reply('Generated timer for session in room ' + channelId + ' was set to ' + sessionStore[channelId].gameState.getCountdownValue());
+    } else {
+      msg.reply('Sorry! A session has not been made yet.  Please type in &highnoon to start a session first.');
+    }
+  });
+
+  ////////////// BELOW IS OLDER IMPLEMENTATION FOR REFERENCE //////////////
```

Comment:
I highly suggest you take this older implementation and move it into a different git branch for storage/keepsake, so that it doesn't clobber your PR.